_posts/2023-09-18-confluent-kafka.md

```diff
@@ -0,0 +1,353 @@
+
+Confluent Cloud uses TLS and SASL. SASL is a username and password. This is the option we'll be using here, and means creating two secrets in Kubernetes, one for the username, and one for the password.
+
+Aiven uses TLS plus a client certificate using their own self-signed CA, which means creating three secrets, one for the CA, one for the client certificate and one for the private key.
+
+The argument that we tend to hear for 1 or 2, is that a team may be running their stack within a private network or VPC. This model only provides the illusion of security, and is not recommended. It can mean that an eavesdropper or malware that is running within the private environment could potentially gain access to the Kafka cluster.
+
+## Provision a Kafka cluster with Confluent Cloud
+
+Head over to [Confluent Cloud](https://www.confluent.io/en-gb/confluent-cloud/) and sign up as a new customer.
+
+Click on Environments and Default, if Default is not displayed, create it.
+
+Click Clusters or "Add Cluster"
+
+![Pick a region, and understand the plan's limits](/images/2023-09-kafka-confluent/create-cluster.png)
+
+At time of writing, the Basic tier of cluster is free, and more than suitable for testing out the Kafka Connector to see how it works.
+
+If you are concerned about being charged by Confluent for your testing, then pay close attention to any limits or quotas that you may exceed.
+
+Pick a cloud from AWS to Google Cloud to Azure, if you already use one of these vendors, use that one so that you can keep your data in the same region.
+
+![Details about your cluster](/images/2023-09-kafka-confluent/created-cluster.png)
+
+### Obtain connection information
+
+Click *Cluster Overview*, then *API Keys* and *Create Key*
+
+For testing, Confluent recommend using the *Global access* key.
+
+Save the Key as "kafka-broker-username.txt"
+
+Save the Secret as "kafka-broker-password.txt"
+
+Add a description such as "openfaas" or "kafka-connector"
+
+Next, click *Cluster Settings, and under *Endpoints* copy the *Bootstrap server* value.
+
+This will look like `pkc-l6wr6.europe-west2.gcp.confluent.cloud:9092`.
+
+Note this down for later user.
+
+Create a topic for testing and name it `faas-request`:
+
+![Create a topic](/images/2023-09-kafka-confluent/topic.png)
+
+For testing, you could set the partition to a smaller value like 3.
+
+Next we'll configure the Helm chart, and use the topic name in the configuration.
+
+## Configure and install the Helm Chart
+
+Add the OpenFaaS Helm chart repository to Helm, then update your repositories:
+
+```bash
+helm repo add openfaas https://openfaas.github.io/faas-netes/
+helm repo update
+```
+
+The [Helm Chart for the Kafka Connector](https://github.com/openfaas/faas-netes/tree/master/chart/kafka-connector) can be installed in two ways:
+
+1) Once per topic - more scalable, with a bit more setup
```

Comment:
A dot should probably be used to render this as a list, e.g `1.`

These are not displaying as a list when I run the site locally.